integration-tests/src/main/java/org/apache/polaris/service/it/test/CatalogFederationIntegrationTest.java

```diff
@@ -0,0 +1,253 @@
+public class CatalogFederationIntegrationTest {
+
+  private static PolarisClient client;
+  private static ManagementApi managementApi;
+  private static PolarisApiEndpoints endpoints;
+  private static ClientCredentials adminCredentials;
+  private static SparkSession spark;
+  private static String sparkToken;
+
+  private static final String PRINCIPAL_NAME = "test-catalog-federation-user";
+  private static final String LOCAL_CATALOG_NAME = "test_catalog_local";
+  private static final String EXTERNAL_CATALOG_NAME = "test_catalog_external";
+  private static final String CATALOG_ROLE_NAME = "catalog_admin";
+  private static final String PRINCIPAL_ROLE_NAME = "service_admin";
+
+  @TempDir static java.nio.file.Path warehouseDir;
+
+  private URI baseLocation;
+  private PrincipalWithCredentials newUserCredentials;
+
+  @BeforeAll
+  static void setup(PolarisApiEndpoints apiEndpoints, ClientCredentials credentials) {
+    endpoints = apiEndpoints;
+    client = polarisClient(endpoints);
+    managementApi = client.managementApi(credentials);
+    adminCredentials = credentials;
+    sparkToken = client.obtainToken(credentials);
+  }
+
+  @AfterAll
+  static void close() throws Exception {
+    if (client != null) {
+      client.close();
+    }
+  }
+
+  @BeforeEach
+  void before() {
+    this.baseLocation = URI.create("file:///tmp/warehouse");
+  }
+
+  @AfterEach
+  void after() {
+    if (spark != null) {
+      SparkSession.clearDefaultSession();
+      SparkSession.clearActiveSession();
+      spark.close();
+    }
+    managementApi.dropCatalog(EXTERNAL_CATALOG_NAME);
+    managementApi.dropCatalog(LOCAL_CATALOG_NAME);
+    managementApi.deletePrincipal(PRINCIPAL_NAME);
+  }
+
+  @Test
+  void testCatalogFederation() {
+    newUserCredentials = managementApi.createPrincipal(PRINCIPAL_NAME);
+    assertThat(newUserCredentials).isNotNull();
+    assertThat(newUserCredentials.getCredentials()).isNotNull();
+    assertThat(newUserCredentials.getCredentials().getClientId()).isNotEmpty();
+    assertThat(newUserCredentials.getCredentials().getClientSecret()).isNotEmpty();
+
+    FileStorageConfigInfo storageConfig =
+        FileStorageConfigInfo.builder()
+            .setStorageType(StorageConfigInfo.StorageTypeEnum.FILE)
+            .setAllowedLocations(List.of(baseLocation.toString()))
+            .build();
+
+    CatalogProperties catalogProperties = new CatalogProperties(baseLocation.toString());
+
+    Catalog localCatalog =
+        PolarisCatalog.builder()
+            .setType(Catalog.TypeEnum.INTERNAL)
+            .setName(LOCAL_CATALOG_NAME)
+            .setProperties(catalogProperties)
+            .setStorageConfigInfo(storageConfig)
+            .build();
+    managementApi.createCatalog(localCatalog);
+    assertThat(managementApi.getCatalog(LOCAL_CATALOG_NAME)).isNotNull();
+    assertThat(managementApi.getCatalog(LOCAL_CATALOG_NAME).getType())
+        .isEqualTo(Catalog.TypeEnum.INTERNAL);
```

Comment:
nit: these look like redundant asserts to be... The test will fail anyway if the catalog is not created. If we want to enhance test coverage of the management API (as is validate that types are not lost on API round-trips) we should probably do that in dedicated (and simpler) tests.

Reply:
Done.